### PR TITLE
Update Dockerfile to create a pub artifact

### PR DESCRIPTION
**Pull request by robbecker-wf**
## Motivation
We want to publish this package to our pub server to remove git dependencies in pubspec.yaml files.

## Changes
Create an artifact for pub in CI so the release pipeline has something to publish.

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/sockjs-dart-client/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/sockjs-dart-client/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: Dockerfile [hunk 1]
```diff
@@ -31,9 +31,10 @@ RUN npm install
 RUN echo "Starting the script section" && \
     pub get && \
     dartanalyzer lib example && \
+    tar czvf sockjs_client.pub.tgz LICENSE README.md pubspec.yaml analysis_options.yaml lib/ && \
```
Comment (seanburke-wf) [inline Dockerfile]: Is there a reason we package analysis_options?

Comment (robbecker-wf) [reply Dockerfile]: it's generally what we should be doing. This way we *could* run proper analysis on the published artifact.

**Comment by seanburke-wf** [comment]
Not much to QA here. Either it will publish or it won't.

**Comment by seanburke-wf** [comment]
@Workiva/release-management-p for merge into master

**Pull request merged**